homeassistant/components/rabbitair/config_flow.py

```diff
@@ -0,0 +1,125 @@
+            with UdpClient(
+                data[CONF_HOST], data[CONF_ACCESS_TOKEN], zeroconf=zeroconf_instance
+            ) as client:
+                info = await client.get_info()
+        except Exception as err:
+            _LOGGER.debug("Connection attempt failed: %s", err)
+            raise
+    except ValueError as err:
+        # Most likely caused by the invalid access token.
+        raise InvalidAccessToken from err
+    except OSError as err:
+        # Most likely caused by the invalid host.
+        raise InvalidHost from err
+    except asyncio.TimeoutError as err:
+        # Either the host doesn't respond or the auth failed.
+        raise TimeoutConnect from err
+    except Exception as err:
+        # Other possible errors.
+        raise CannotConnect from err
+
+    # Return info to store in the config entry.
+    return {"title": f"RabbitAir-{info.mac.replace(':', '')}", "mac": info.mac}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Rabbit Air."""
+
+    VERSION = 1
+
+    _discovered_host: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAccessToken:
+                errors["base"] = "invalid_access_token"
+            except InvalidHost:
+                errors["base"] = "invalid_host"
+            except TimeoutConnect:
+                errors["base"] = "timeout_connect"
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.debug("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(dr.format_mac(info["mac"]))
+                self._abort_if_unique_id_configured(updates=user_input)
+                return self.async_create_entry(title=info["title"], data=user_input)
+
+        user_input = user_input or {}
+        host = user_input.get(CONF_HOST, self._discovered_host)
+        token = user_input.get(CONF_ACCESS_TOKEN)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=host): str,
+                    vol.Required(CONF_ACCESS_TOKEN, default=token): vol.All(
+                        str, vol.Length(min=32, max=32)
+                    ),
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        mac = dr.format_mac(discovery_info.properties["id"])
```

Comment:
Same as above, there is no need to mac format the unique ID for a config entry.

Reply:
Removed the format_mac call. But formatting is still required, because different forms of MAC-address representation are possible. In this case, the string received from Zeroconf does not match the string returned by the device API.

Reply:
I discussed this with Frenck, the documentation is indeed correct.
Please revert back to calling `format_mac`